Update callbacks. (1) are they invoked, and (2) is the record fully updated?

caterwaul.clone('std', 'error', 'iter', 'db.file')(function () {
  var db = this[caterwaul].db.file('test-db'), n = 0;

  let[r = db('update-callbacks')] in iter.n[i, 1000][r.update('bif', i, fn_[r.get('bif', fn[x][++n, error.fail['Wrong value: #{x} !== 1000'], unless[x === 1000]])])];
  setTimeout(fn_[n === 1000 || error.fail['Wrong value for n: #{n} !== 1000']], 1000);
})();
